test/kaniko_task_test.go

```diff
@@ -89,12 +116,40 @@ func getTask(namespace string, t *testing.T) *v1alpha1.Task {
 					Name:  "kaniko",
 					Image: "gcr.io/kaniko-project/executor",
 					Args: []string{"--dockerfile=/workspace/Dockerfile",
-						fmt.Sprintf("--destination=%s/kanikotasktest", dockerRepo),
+						fmt.Sprintf("--destination=%s", repo),
 					},
 				}},
 			},
 		},
 	}
+	if withSecretConfig {
+		step := task.Spec.BuildSpec.Steps[0]
+		step.VolumeMounts = []corev1.VolumeMount{
+			{
+				Name:      "kaniko-secret",
+				MountPath: "/secrets",
+			},
+		}
+		step.Env = []corev1.EnvVar{
+			{
+				Name:  "GOOGLE_APPLICATION_CREDENTIALS",
+				Value: "/secrets/config.json",
```

Comment:
one thing im not a super fan of is that this is a very google specific solution (not that i have anything better to suggest!) - but [looking at the kaniko docs](https://github.com/GoogleContainerTools/kaniko#kubernetes-secret) i guess this is the only option?

i'm guessing this would require that anyone running this test have gcloud installed so that `/secrets/config.json` gets created? (would be good to include this info in test/README.md and/or DEVELOPMENT.md as [a requirement](https://github.com/knative/build-pipeline/blob/master/DEVELOPMENT.md#requirements))

Reply:
Yup, the only thing I could find was the google_app_cred auth method.

For config.json - I created a GCP service account and exported a JSON key. I could write a script that would provision a service account for you outside of the e2e tests with the proper IAM permissions, but that would indeed require gcloud.